go/pkg/reader/reader_test.go

```diff
@@ -0,0 +1,127 @@
+package reader
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func setTmpFile(t *testing.T, content []byte) (string, func()) {
+	t.Helper()
+	execRoot, err := ioutil.TempDir("", "")
```

Comment:
nit: We already have something like this in filemetadata_test.go. We should just centralize it in some test util package. Also, no real need to clean up temp files.

Reply:
Rebased this on top of newly added #234.